### ISSUE TITLE
RET504 potential false-positive

### ISSUE THREAD

**Issue by twoertwein** [charliermarsh/ruff#1606]
flake8-return behaves the same as ruff in the following example:

```py
from pathlib import Path

def add_extension(path: Path, extension: str, *, has_to_exists: bool = False):
    new_path = path.with_suffix(f".{extension}")
    if has_to_exists and not new_path.is_file() and path.is_file():
        new_path = path
    return new_path
```
> test.py:8:12: RET504 Unnecessary variable assignment before `return` statement

I assume flake8-return would prefer an (ugly) `return path if (..........) else .....`. Is this the intended behavior, or should a second assignment to the return variable be counted as a usage?

**Comment by charliermarsh** [comment]
I think it wants:

```py
from pathlib import Path

def add_extension(path: Path, extension: str, *, has_to_exists: bool = False):
    new_path = path.with_suffix(f".{extension}")
    if has_to_exists and not new_path.is_file() and path.is_file():
        return path
    return new_path
```
